File: aggregate.py
import sys
import pickle

# aggregate all the trained models
models = []
for line in sys.stdout.buffer:
    models.append(pickle.load(line))

# Train global data using federated learning
# ...

# serialize global model
global_model_data = pickle.dumps(global_model)

# send the serialized data to train.c
sys.stdout.buffer.write(global_model_data)
sys.stdout.flush()

File: aggregate.c
#include<stdio.h>
#include<stdlib.h>
#include<unistd.h>

int main(){
    // recieve serialized global model from aggregate.py
    char buffer[1024]; // should it really be this small ??
    int count = read(STDIN_FILENO, buffer, sizeof(buffer));

    // send serialized global model to train.py
    FILE* fp = popen("python3 train.py" , "w");
    fwrite(buffer, 1, count, fp);
    pclose(fp);

    return 0;
}